src/routes/default.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * AR.IO Gateway
+ * Copyright (C) 2022-2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Router } from 'express';
+import { APEX_TX_ID, APEX_ARNS_NAME, ARNS_ROOT_HOST } from '../config.js';
+import { dataHandler } from './data/index.js';
+import { arnsMiddleware } from './arns.js';
+import { arIoInfoHandler } from './ar-io.js';
+
+export const defaultRouter = Router();
+
+defaultRouter.get('/', async (req, res, next) => {
+  if (APEX_TX_ID === undefined && APEX_ARNS_NAME === undefined) {
+    return arIoInfoHandler(req, res);
+  }
+
+  const modifiedReq = Object.create(req);
+
+  if (APEX_TX_ID !== undefined) {
+    modifiedReq.params = { 0: APEX_TX_ID, ...req.params };
+    return dataHandler(modifiedReq, res, next);
+  }
+
+  if (APEX_ARNS_NAME !== undefined) {
+    // Use Proxy to modify hostname as it is read only
+    const modifiedReq = new Proxy(req, {
```

Comment:
modifiedReq is already in scope, declared as const, is this a legal operation?

Reply:
Yes. The `if` will creates a new scope. That said, I think the earlier declaration could be moved inside the `APEX_TX_ID` conditional for clarity.

Reply:
[`0312eea` (#344)](https://github.com/ar-io/ar-io-node/pull/344/commits/0312eea635037077707b3401c106c7767a17b45f)